app/assets/javascripts/thyme/components/annotations_toggle.js

```diff
@@ -0,0 +1,53 @@
+class AnnotationsToggle extends Component {
+
+  constructor(element) {
+    super(element);
+    this.id = element;
+    this.check = document.getElementById(this.id + '-check');
+    this.checkJQ = $('#' + this.id + '-check');
+    this.div = $('#' + this.id);
+  }
+
+  add() {
+    const toggle = this;
+    const checkJQ = this.checkJQ;
+    
+    /* User is teacher/editor for the given medium and visible_for_teacher ist activated?
+       -> add toggle annotations button */
+    $.ajax(Routes.check_annotation_visibility_path(thymeAttributes.mediumId), {
+      type: 'GET',
+      dataType: 'json',
+      success: function(isPermitted) {
+        if (isPermitted) {
+          toggle.show();
+          toggle.element.addEventListener('click', function() {
+            thymeAttributes.annotationManager.updateAnnotations(toggle.getValue());
+          });
+          // When loading the player, the toggle is set to "true" by default,
+          // so we have to trigger updateAnnotations() manually once.
+          thymeAttributes.annotationManager.updateAnnotations(toggle.getValue());
+        }
+      }
+    });
+  }
+
+  /*
+    Returns true if the toggle's value is true and false otherwise.
+   */
+  getValue() {
+    return this.checkJQ.is(":checked");
+  }
+
+  /*
+    Auxiliary method
+  */
+  show() {
+    $('#volume-controls').css('left', '66%');
+    $('#speed-control').css('left', '77%');
+    $('#emergency-button').css('left', '86%');
+    thymeAttributes.hideControlBarThreshold.x = 960;
+    this.div.show();
+    //updateControlBarType(); TODO
```

Comment:
Good catch!

I think, it's not necessary to update the control bar type here (any more), because the updateControlBarType() method is called at the very end of the thyme player script (when everything is loaded, and it should stay like this way).